### PR TITLE
Add/stats page upsells actions

### PR DESCRIPTION
**Pull request by leogermani**
This is the second step in the Upgrade Nudges on the Stats page.

This PR adds the button actions and tracking.

#### Changes proposed in this Pull Request:

* Add links to the buttons
* Tracks the events when buttons are clicked

#### Jetpack product discussion

p1HpG7-cfa-p2

#### Does this pull request change what data or activity we track or use?

We track the events when the new Upgrade Buttons on the Stats page are clicked

#### Testing instructions:

We are going to test 2 things: Tracking and actions

For all tests:

* Connect your site
* Edit your `wp-config.php` and add `define( 'JETPACK_DEV_TEST_STATS_UPGRADE_NUDGES', true );`
* Got to Jetpack > Site Stats

Testing tracking:

* You should see all 7 items on the bottom of the page. Click all of them and check Tracks live view for the following events: `jetpack_stats_nudges_security`, `jetpack_stats_nudges_scan`, `jetpack_stats_nudges_backup`, `jetpack_stats_nudges_akismet`, `jetpack_stats_nudges_search`, `jetpack_stats_nudges_crm`, and`jetpack_stats_nudges_boost`.

Testing the actions:

You'll need to test all the actions below with a site fully connected and then with a site connected only at a site level.

Security -> should take you to the Plans page. Make the purchase and confirm the plan is added to your site and you are redirected back to WP-Admin

Backup -> should take you to the Backup landing page. Make the purchase and confirm the plan is added to your site and you are redirected back to WP-Admin

Scan -> should take you to the Scan landing page. Make the purchase and confirm the plan is added to your site and you are redirected back to WP-Admin

Anti-spam -> should take you to the Anti-spam landing page. Make the purchase and confirm the plan is added to your site and you are redirected back to WP-Admin

Search -> should take you to the Search landing page. Make the purchase and confirm the plan is added to your site and you are redirected back to WP-Admin (NOTE: Search will not work for a site-only connection. We either have to make it work or hide this nudge. See p1626445219079300-slack-CBG1CP4EN

### CODE REVIEW

File: projects/plugins/jetpack/modules/stats/class-jetpack-stats-upgrade-nudges.php [hunk 1]
```diff
@@ -189,27 +305,22 @@ public static function render( $callback ) {
...
+			self::print_security();
 			if ( ! self::is_backup_active() ) {
-				self::print_item( __( 'Backup', 'jetpack' ), 'products/product-jetpack-backup.svg', true );
+				self::print_backup();
 			}
 			if ( ! self::is_scan_active() ) {
-				self::print_item( __( 'Scan', 'jetpack' ), 'products/product-jetpack-scan.svg', true );
+				self::print_scan();
 			}
 			if ( self::is_akismet_active() ) {
```
Comment (kbrown9) [inline projects/plugins/jetpack/modules/stats/class-jetpack-stats-upgrade-nudges.php:315]: We want to check whether Akismet is not active:

```suggestion
			if ( ! self::is_akismet_active() ) {
```

**Comment by kbrown9** [comment]
We[ just check](https://github.com/Automattic/jetpack/blob/master/projects/plugins/jetpack/modules/stats/class-jetpack-stats-upgrade-nudges.php#L97) whether the `Akismet_Admin` class exists when determining whether to display the Anti-Spam item. I'm wondering if that's a sufficient check. We might want to also check whether the site has an Akismet API key and, if it doesn't, display the Anti-Spam item.

**Comment by kbrown9** [comment]
Ideally, we would only render the header (and add the styles) if we have product items to display:

Screen%20Shot%20on%202021-07-21%20at%2011-20-44

**Comment by kbrown9** [comment]
I tested the current changes, and everything worked as expected. For all of the items, the button redirected to the correct purchase page or installed/activated the correct plugin. The tracks events were generated as expected.

**Comment by leogermani** [comment]
@kbrown9 thanks for the review!

The last commits address the issues you raised:

* Do not print the header and styles if there is nothing to show - https://github.com/Automattic/jetpack/pull/20381/commits/d6696b6e535cc3f91a6b2b434072e5fca180644b
* If user has scan, backups and anti-spam, do not show Security - https://github.com/Automattic/jetpack/pull/20381/commits/654c0b9aa87e237f712ebe47845873f70b80ab88 (p1HpG7-cfa-p2#comment-47894)
* Change the title of the Section if user has Security - https://github.com/Automattic/jetpack/pull/20381/commits/72bbd8f63da652ddd0a46b8b1cb748eef95446fd (p1HpG7-cfa-p2#comment-47893)
* Consider if user has Akismet Key when checking if Akismet is active - https://github.com/Automattic/jetpack/pull/20381/commits/0b2c33a9b48db1a9b2ca85310980b362e7630fc9
* Don't display the whole section if user has Complete plan - https://github.com/Automattic/jetpack/pull/20381/commits/bbd0b54cd0b59c948041666dd0f1a1e440240d18 (p1HpG7-cfa-p2#comment-47893)

As for the edge case with Akismet, I think it's fine for this Proof of Concept.

Note: I didn't add any memorization to the methods because all of them have some kind of caching on any expensive operations, so we should be fine. But let me know what you think.

File: projects/plugins/jetpack/modules/stats/class-jetpack-stats-upgrade-nudges.php [hunk 2]
```diff
@@ -178,38 +326,51 @@ private static function print_item( $title, $icon, $subitem = false, $button_lab
...
+		if (
+			self::has_security_plan() &&
+			self::is_backup_active() &&
+			self::is_scan_active() &&
+			self::is_akismet_active() &&
+			self::is_search_active() &&
+			self::is_backup_active() &&
+			self::is_backup_active() &&
+			self::get_boost_output( false ) &&
+			self::get_crm_output( false )
```
Comment (kbrown9) [inline projects/plugins/jetpack/modules/stats/class-jetpack-stats-upgrade-nudges.php:350]: It looks like we already check `is_backup_active` in the second line of the if conditions. Also, we want to check that `get_boost_output` and `get_crm_output` return false.

```suggestion
			! self::get_boost_output( false ) &&
			! self::get_crm_output( false )
```

**Review by kbrown9 — APPROVED** [review]
This tests well for me. The buttons display and function as expected, and the tracks events are recorded as expected.

There are a few things I'd like to clarify with the design team:
* Should we display the Boost and CRM items with a Complete Plan?
* Should we customize the heading text based on whether the Search, Boost, and CRM items are displayed, like we're customizing it based on whether the Security items are displayed?

These issues are low priorities. If we want to address them, we can do that in a separate PR.

**Comment by leogermani** [comment]
Thank you very much Kim! And sorry for the lack of attention to the details there :( 

I'll merge it now and work on the styling

**Pull request merged**